ladybug/windrose.py

```diff
@@ -0,0 +1,156 @@
+                 ):
+        """
+        Instantiates the Ladybug WindRose Object
+        """
+        self.epw = EPW(epw_filePath)
+        self.HOYs = HOYs
+        self.annualHourlyData = annualHourlyData
+        self.windCondition = self.check_condition(windCondition)
+        self.dataCondition = self.check_condition(dataCondition)
+
+    @staticmethod
+    def check_condition(condition):
+        """
+        Checks whether the conditional statement has
+        the variable 'x'
+        Args:
+            condition: A list of conditional statements
+        Returns:
+            None: if condition is None or the variable 'x' is not \
+            in the conditional statement
+            condition: The same list of conditional statements provided
+            as arguments to the function
+
+        """
+        if condition is None:
+            return None
+        if isinstance(condition, str):
+            if "x" in condition.lower():
+                return condition
+            else:
+                print ("The conditon shall be a valid string with" +
+                       " a variable 'x' in it.")
+        else:
+            xcount = 0
+            for statement in condition:
+                if "x" in statement.lower():
+                    xcount += 1
+            if len(condition) == xcount:
+                return condition
+            else:
+                print ("All conditions shall be a valid string with" +
+                       "a variable 'x' in it.")
+                return None
+
+    def data_to_HOY(self):
+        """
+        This function takes the list of annual hourly data
+        and the list of data conditions. By taking them into consideration
+        this function creates a list of HOYs that can be used to
+        craft the wind data
+        Args:
+            The instance of the WindRose Object
+        Returns:
+            A sorted list of common HOYs as per provided
+            annualHourlyData and the dataCondition. It is
+            the list of HOYs that pass all the dataConditions
+            provided.
+        """
+        if not len(self.annualHourlyData) == len(self.dataCondition):
+            print ("Then number of condtional statements don't match" +
+                   " the number of annualHourlyData provided." +
+                   " Therefore, conditional statements are ignored")
+            return None
+        else:
+            # The following is a list of lists
+            # It catches HOYs for all the annualHourlyData
+            # dataCondition pairs
+            HOYList = []
+            for i in range(len(self.annualHourlyData)):
+                HOYList.append(
+                    self.annualHourlyData[i].data_to_HOY(
+                        self.dataCondition[i]))
+            HOYSets = [set(listItem) for listItem in HOYList]
+            HOYCommon = set.intersection(*HOYSets)
+            return sorted(HOYCommon)
+
+    def parse_wind_data(self):
```

Comment:
This method can be broken down into several properties. Let's think about the most common properties that one wants to get from a WindRose. Frequency for each direction? Average wind speed for each direction? Then each of them should be accessible directly. If you haven't watched this talk already I think you should watch it before revising the code here: https://www.youtube.com/watch?v=OSGv2VnC0go&t=9s